tls-client/main.cpp

```diff
@@ -216,23 +219,21 @@ class HelloHTTPS {
 
        /* Start the handshake, the rest will be done in onReceive() */
         mbedtls_printf("Starting the TLS handshake...\r\n");
-        ret = mbedtls_ssl_handshake(&_ssl);
+        do {
+            ret = mbedtls_ssl_handshake(&_ssl);
+        } while (ret != 0 && (ret == MBEDTLS_ERR_SSL_WANT_READ ||
+                ret == MBEDTLS_ERR_SSL_WANT_WRITE));
         if (ret < 0) {
-            if (ret != MBEDTLS_ERR_SSL_WANT_READ &&
-                ret != MBEDTLS_ERR_SSL_WANT_WRITE) {
-                print_mbedtls_error("mbedtls_ssl_handshake", ret);
-                onError(_tcpsocket, -1 );
-            }
+            print_mbedtls_error("mbedtls_ssl_handshake", ret);
```

Comment:
member ```_error```, functions ```error()``` and ```done``` aren't used anywhere. Hence setting it also doesn't make sense. looks more like incomplete code. I would rather remove it.